src/includes/service.h

```diff
@@ -458,7 +459,7 @@ class service_record
     // any appropriate cleanup.
     virtual void becoming_inactive() noexcept { }
 
-    // Check whether the service should automatically restart (assuming auto_restart is true)
+    // Check whether the service should automatically restart (assuming auto_restart is ALWAYS)
```

Comment:
The comment implies that this function isn't used if auto_restart is ON_FAIL, but that can't be right?

Reply:
Yes, It wasn't right. The right thing is `(assuming auto_restart is ALWAYS or ON_FAILURE)`